03_Collection Types.playground/section-1.swift

```diff
@@ -18,6 +18,7 @@ import UIKit
 
 // Arrays and dictionaries in swift use generics and can be mutable or immutable
 // depending on whether they are assigned to a var or let
+// they are VALUES type and a STRUCT, so must be assigned to new variable when using non mutating internal function
```

Comment:
Here's a blog post that talks about what you're referring to.
[https://www.natashatherobot.com/mutating-functions-swift-structs/](url)

> Remember, a Struct is a value type. In other words, it’s supposed to be immutable. That’s confusing, considering it appears like we’ve added a function that mutates the internals of our struct. Yep, the mutating keyword is starting to make sense now.

> By having to make our struct a variable instead of a constant to use the mutating function, Swift is giving us a very strong hint as to what is happening. When we call our mutating scaleBy function, it actually replaces the original rectangle stored in the myRect variable with a brand new rectangle (with the new scaled width and height in our case).